modules/nf-core/gatk4/haplotypecaller/main.nf

```diff
@@ -29,6 +30,7 @@ process GATK4_HAPLOTYPECALLER {
     def dbsnp_command = dbsnp ? "--dbsnp $dbsnp" : ""
     def interval_command = intervals ? "--intervals $intervals" : ""
     def dragstr_command = dragstr_model ? "--dragstr-params-path $dragstr_model" : ""
+    def bamout_command = args.contains("--bam-writer-type") ? "--bam-output ${prefix.replace('.g', '')}.realigned.bam" : ""
```

Comment:
Isn't the `.g` suffix highly pipeline specific?

Reply:
nope, it's pretty common actually

Reply:
but only for vcf, this way, I'm that it's not propagated to the bam file even if specified for the vcf one

Reply:
To be fair, the perfect solution would be to have 2 prefixes, but that sounds like crazy

Reply:
True that!

Reply:
or if `prefix` is something file `name.genomex`

Reply:
it does just replace `.g` by nothing, could be more efficient by removing `.g` if it's at the end of the string indeed, that would solve this `sample.gnarly_name` issue

Reply:
now fixed